packages/langium/src/grammar/lsp/langium-grammar-reference-finder.ts

```diff
@@ -0,0 +1,196 @@
+        const groupNode = getContainerOfType(targetAstNode, isGroup);
+
+        if (parserRule?.returnType?.ref) {
+            const interfaces = this.getInterfacesFromAssignment(parserRule.returnType.ref);
+            interfaces.forEach(interfaceNode => {
+                const leafNode = this.findLeafNodeInInterface(interfaceNode,selectedNode.text);
+                if (leafNode) {
+                    refs.push(...this.getReferences(leafNode, params, document));
+                }
+            });
+        }
+
+        if (groupNode) {
+            const action = groupNode.elements.find(el => isAction(el));
+            if (isAction(action)) {
+                const ref = action.type?.ref;
+                if (ref) {
+                    const interfaces = this.getInterfacesFromAssignment(ref);
+                    interfaces.forEach(interfaceNode => {
+                        const leafNode = this.findLeafNodeInInterface(interfaceNode,selectedNode.text);
+                        if (leafNode) {
+                            refs.push(...this.getReferences(leafNode, params, document));
+                        }
+                    });
+                }
+
+            }
+        }
+
+        return refs;
+    }
+
+    findLeafNodeInInterface(interfaceNode: Interface, text: string): LeafCstNode | undefined {
+        const attribute = interfaceNode.attributes.find(a => a.name === text);
+        if (attribute) {
+            const cstNode = attribute.$cstNode;
+            if (cstNode) {
+                const leafNode = findLeafNodeAtOffset(cstNode, cstNode.offset);
+                if (leafNode) {
+                    return leafNode;
+                }
+            }
+        }
+        return undefined;
+    }
+
+    getInterfacesFromAssignment(ref: AbstractType): Set<Interface> {
+        let interfaces: Set<Interface> = new Set<Interface>();
+        if (isInterface(ref)) {
+            interfaces = this.collectSuperTypes(ref);
+            interfaces.add(ref);
+        } else if (isType(ref)) {
+            interfaces = this.collectSuperTypes(ref);
+        }
+        return interfaces;
+    }
+
+    getReferencesFromTypeAttribute(typeAttributeNode: TypeAttribute, selectedNode: LeafCstNode): Array<{ docUri: URI; range: Range; }> {
+        const refs: Array<{ docUri: URI; range: Range; }> = [];
+        const interfaceNode = getContainerOfType(typeAttributeNode, isInterface);
+        if (interfaceNode) {
+            const collectedTypes = this.collectChildrenTypes(interfaceNode);
+            collectedTypes.add(interfaceNode);
+
+            let referencesToTypes: Set<ReferenceDescription> = new Set<ReferenceDescription>();
+            collectedTypes.forEach(collectedType => {
+                const refs = this.references.findReferences(collectedType).toSet();
+                referencesToTypes = new Set<ReferenceDescription>([...referencesToTypes, ...refs]);
+            });
```

Comment:
Minor: I think there might be a lot of unnecessary computation in there (constructing multiple sets again and again).

```suggestion
            const referencesToTypes = new Set<ReferenceDescription>();
            collectedTypes.forEach(collectedType => {
                const refs = this.references.findReferences(collectedType);
                for (const ref of refs) {
                    referencesToTypes.add(ref);
                }
            });
```